src/data/java/com/github/minecraftschurlimods/arsmagicalegacy/data/AMRecipeProvider.java

```diff
@@ -480,7 +480,7 @@ protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
                 .requires(Tags.Items.CROPS_POTATO)
                 .requires(Items.ICE)
                 .requires(Items.SUGAR)
-                .requires(NBTIngredient.of(PotionUtils.setPotion(new ItemStack(Items.POTION), Potions.WATER)))
+                .requires(StrictNBTIngredient.of(PotionUtils.setPotion(new ItemStack(Items.POTION), Potions.WATER)))
```

Comment:
Should probably be PartialNBTIngredient instead, StrictNBTIngredient does not allow additional NBT data (such as names or potential additional NBT from other mods). Goes for all usages of NBTIngredient in this file.

Reply:
PartialNBTIngredient has different parameters (Item, CompoundTag) but PotionUtils.setPotion needs and gives an ItemStack

Reply:
What about just doing this:
```java
PartialNBTIngredient.of(Items.POTION, PotionUtils.setPotion(new ItemStack(Items.POTION), Potions.WATER).getTag())
```
(or whatever the method for getting NBT from an ItemStack is called)